src/fetch_request.js

```diff
@@ -35,14 +38,18 @@ export class FetchRequest {
   }
 
   get fetchOptions () {
-    return {
+    const requestOptions = {
       method: this.method.toUpperCase(),
       headers: this.headers,
       body: this.body,
       signal: this.signal,
       credentials: 'same-origin',
       redirect: 'follow'
     }
+
+    this._stringifyBody(requestOptions)
+
+    return requestOptions
```

Comment:
rather than introducting a semi private (underscore) function that is not really in use currently in this project, an alternative could be to have a new getter.

```js 
get fetchOptions () {
  return {
    method: this.method.toUpperCase(),
    headers: this.headers,
    body: this.formatedBody,
    signal: this.signal,
    credentials: 'same-origin',
    redirect: 'follow'
  }
}


....

get formatedBody() {
  const bodyIsAString =
    Object.prototype.toString.call(this.body) === "[object String]";
  const contentTypeIsJson =
    this.headers["Content-Type"] === "application/json";

  if (contentTypeIsJson && !bodyIsAString) {
    return JSON.stringify(this.body);
  } else {
    return this.body;
  }
}
```

Reply:
That looks better 👍 did it